app/models/grda_warehouse/hud/client.rb

```diff
@@ -326,11 +326,12 @@ class Client < Base
         select(:client_id).distinct)
     end
 
-    scope :has_homeless_service_between_dates, ->(start_date: 31.days.ago, end_date: Date.current) do
-      where(id:
-        GrdaWarehouse::ServiceHistoryService.homeless(chronic_types_only: true).
-        where(date: (start_date..end_date)).
-        select(:client_id).distinct)
+    scope :has_homeless_service_between_dates, ->(start_date: 31.days.ago, end_date: Date.current, include_extrapolated: true) do
+      shs_query = GrdaWarehouse::ServiceHistoryService.homeless(chronic_types_only: true).
+        where(date: (start_date..end_date))
+      return where(id: shs_query.select(:client_id).distinct) if include_extrapolated
+
+      where(id: shs_query.service_excluding_extrapolated.select(:client_id).distinct)
```

Comment:
I found the multiple exits a little confusing... Maybe this is simpler?

```suggestion
    scope :has_homeless_service_between_dates, ->(start_date: 31.days.ago, end_date: Date.current, include_extrapolated: true) do
      shs_query = GrdaWarehouse::ServiceHistoryService.homeless(chronic_types_only: true).
        distinct.
        where(date: (start_date..end_date))
     shs_query = shs_query.service_excluding_extrapolated unless include_extrapolated
      where(id: shs_query.select(:client_id))
```